naresh/src/main/java/com/lftechnology/phpjava/assignmentthree/LargestPalindrome.java

```diff
@@ -0,0 +1,78 @@
+package com.lftechnology.phpjava.assignmentthree;
+
+
+import com.lftechnology.phpjava.menu.MenuReader;
+import com.lftechnology.phpjava.menu.ProgramMenu;
+
+import java.util.Scanner;
+
+/**
+ * @author Naresh Maharjan <nareshmaharjan@lftechnology.com>
+ */
+public class LargestPalindrome implements MenuReader {
+
+    /**
+     * find the largest palindrome number
+     *
+     * @param startIndex
+     * @param endIndex
+     * @return
+     * @author Naresh Maharjan <nareshmaharjan@lftechnology.com>
+     */
+    public long findLargestPalindrome(int startIndex, int endIndex) {
+        long palindrome = 0;
+        int max = Math.max(startIndex, endIndex);
+        int min = Math.min(startIndex, endIndex);
+        for (int i = max; i >= min; i--) {
+            for (int j = max; j >= min; j--) {
+                long product = i * j;
+                if (palindrome < product && isPalindrome(product)) {
+                    palindrome = product;
+                }
+            }
+        }
+        return palindrome;
+    }
+
+    /**
+     * check if the number is palindrome or not
+     *
+     * @param nr
+     * @return
+     * @author Naresh Maharjan <nareshmaharjan@lftechnology.com>
+     */
+    public boolean isPalindrome(long nr) {
+        long rev = 0;                    // the reversed number
+        long x = nr;                     // store the default value (it will be changed)
+        while (x > 0) {
+            rev = 10 * rev + x % 10;
+            x /= 10;
+        }
+        return nr == rev;               // returns true if the number is palindrome
+    }
+
+    /**
+     * read the user input for the action
+     */
+    @Override
+    public void scanner() {
+        System.out.println("Please enter two numbers to find out the largest palindrome possible from the its product.");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.java.lftechnology.com:9000/coding_rules#rule_key=squid%3AS106)